### PR TITLE
[SPARK-11484][WebUi] Using proxyBase set by spark AM

### PR DESCRIPTION
**Pull request by vundela**
Use the proxyBase set by the AM, if not found then use env. This is to fix the issue if somebody accidentally set APPLICATION_WEB_PROXY_BASE to wrong proxyBase

### CODE REVIEW

**Comment by vanzin** [comment]
ok to test

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala [hunk 1]
```diff
@@ -143,12 +143,12 @@ private[spark] object UIUtils extends Logging {
 
   // Yarn has to go through a proxy so the base uri is provided and has to be on all links
   def uiRoot: String = {
-    if (System.getenv("APPLICATION_WEB_PROXY_BASE") != null) {
-      System.getenv("APPLICATION_WEB_PROXY_BASE")
-    } else if (System.getProperty("spark.ui.proxyBase") != null) {
+    // SPARK-11484 - Use the proxyBase set by the AM, if not found then use env.
+    if (System.getProperty("spark.ui.proxyBase") != null) {
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala]: I'd use the opportunity to clean this up a bit:

    sys.props("spark.ui.proxyBase").orElse(sys.env("APPLICATION_WEB_PROXY_BASE")).getOrElse("")

**Comment by vanzin** [comment]
LGTM. My only concern is that someone might be using yarn-client mode with a different UI proxy than the RM; this change makes that impossible, since the AM always sets `spark.ui.proxyBase`. I find that case rather unlikely though.

**Comment by vundela** [comment]
Thanks for the review @vanzin, made the changes as you suggested.

**Comment by vanzin** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/ui/UIUtils.scala [hunk 2]
```diff
@@ -143,14 +143,10 @@ private[spark] object UIUtils extends Logging {
...
   def uiRoot: String = {
-    if (System.getenv("APPLICATION_WEB_PROXY_BASE") != null) {
-      System.getenv("APPLICATION_WEB_PROXY_BASE")
-    } else if (System.getProperty("spark.ui.proxyBase") != null) {
-      System.getProperty("spark.ui.proxyBase")
-    }
-    else {
-      ""
-    }
+    // SPARK-11484 - Use the proxyBase set by the AM, if not found then use env.
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala]: I think this ultimately makes sense, letting Spark-specific stuff take precedence, though it's worth enumerating here or in the JIRA why this is connected to Oozie / uber mode, and why this is right-er than the current behavior.

Comment (vundela) [inline core/src/main/scala/org/apache/spark/ui/UIUtils.scala]: Thanks for the review @srowen. Updated the JIRA to reflect the customer reported issue and why it is not related to Oozie/uber mode.

**Comment by vanzin** [comment]
Merging to master.

**Pull request closed without merging**